rust/transaction-filter/src/filters/move_module.rs

```diff
@@ -0,0 +1,76 @@
+use crate::traits::Filterable;
+use anyhow::{anyhow, Error};
+use aptos_protos::transaction::v1::{EntryFunctionId, MoveModuleId, MoveStructTag};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct MoveModuleFilter {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub address: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+}
+
+impl Filterable<MoveModuleId> for MoveModuleFilter {
+    fn is_valid(&self) -> Result<(), Error> {
+        if self.address.is_none() && self.name.is_none() {
+            return Err(anyhow!("At least one of address or name must be set"));
+        };
+        Ok(())
+    }
+
+    fn is_allowed(&self, module_id: &MoveModuleId) -> bool {
+        self.address.is_allowed(&module_id.address) && self.name.is_allowed(&module_id.name)
+    }
+}
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct EntryFunctionFilter {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub module: Option<MoveModuleFilter>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub method: Option<String>,
```

Comment:
Let's go with `function`, it's what they call em in Move.

Reply:
un-nested into:
```
#[derive(Debug, Deserialize, PartialEq, Serialize)]
#[serde(deny_unknown_fields)]
pub struct EntryFunctionFilter {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub address: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub name: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub method: Option<String>,
}
```

will rename